pkg/controller/uniteddeployment/uniteddeployment_controller.go

```diff
@@ -231,6 +233,14 @@ func (r *ReconcileUnitedDeployment) Reconcile(_ context.Context, request reconci
 		return reconcile.Result{}, err
 	}
 
+	selector, err := util.ValidatedLabelSelectorAsSelector(instance.Spec.Selector)
```

Comment:
just using ValidatedLabelSelectorAsSelector is not enough to identify some label selector errors, you can call LabelSelectorAsSelector and check if there is any error. Similar code is in the cloneset validation validateCloneSetSpec

Reply:
Do you mean validate UnitedDeploymentSpec? https://github.com/openkruise/kruise/blob/9e74506ca800877f5bf63723efa33521a3533c46/pkg/webhook/uniteddeployment/validating/uniteddeployment_validation.go#LL52C60-L52C60

Reply:
yes, it seems the uniteddeployment webhook already validate the selector, is it necessary to validate again the controller

Reply:
LabelSelectorAsSelector has been called in the validateUnitedDeploymentSpec method to identify some label selector errors.
I think ValidatedLabelSelectorAsSelector is ok here.

Reply:
> yes, it seems the uniteddeployment webhook already validate the selector, is it necessary to validate again the controller

Unnecessary, i think.